lib/lib/layer.lib.ts

```diff
@@ -23,17 +23,20 @@ export type LayersWithSource =
   | RasterLayerSpecification
   | HillshadeLayerSpecification;
 
-export type LayerProps = Omit<
-  LayersWithSource,
-  "source" | "source-layer" | "id" | "type"
-> & {
-  layerId?: string;
+export type LayersWithoutSource = BackgroundLayerSpecification;
```

Comment:
This type is not used. Should we keep it?

Reply:
I thought about that - since we don't have a function that uses these types of layers generically right now it it doesn't get use anywhere.

I do think it is helpful for maintainers as documentation to indicate that there are in fact layers that don't require sources, which wasn't apparent to me on first reading the code, and using the background layers super often in my code.

Since it is a type, it should get stripped out at build time and not increase the bundle size for real applications.

If you feel strongly about it though I can take it out!

Reply:
Make sense. Thanks for your work.